src/generator/AutoRest.Ruby.Azure/Model/CompositeTypeRba.cs

```diff
@@ -42,15 +50,49 @@ public override string GetBaseTypeName()
                 if (this.BaseModelType.Extensions.ContainsKey(AzureExtensions.ExternalExtension) ||
                     this.BaseModelType.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension))
                 {
-                    typeName = "MsRestAzure::" + typeName;
+                    if(!nameRegEx.IsMatch(typeName) || !IsResourceModelsMatchStandardDefinition(this))
+                    {
+                        typeName = "MsRestAzure::" + typeName;
+                    }
                 }
 
                 return " < " + typeName;
             }
+            else if (nameRegEx.IsMatch(this.Name.ToString()))
+            {
+                return " < " + "MsRestAzure::" + this.Name.ToString();
+            }
 
             return string.Empty;
         }
 
+        public static bool IsResourceModelsMatchStandardDefinition(CompositeType model)
+        {
+            string modelName = model.Name.ToString();
+            if (
+                (modelName.Equals("SubResource", StringComparison.InvariantCultureIgnoreCase) &&
+                    model.Properties.All(property => subResourceRegEx.IsMatch(property.Name.ToString()))) ||
```

Comment:
is there a way to query the properties of the actual class/model, instead of hardcoding the subResourceRegEx and resourceRegEx?